<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>01-promise的基本使用</title>
</head>
<body>



<script>

  //参数->函数(resolve, reject)
  // resolve, reject本身它们又是函数
  // new Promise((resolve, reject) => {
  //   setTimeout(()=>{
  //     console.log('Hello world!');
  //     console.log('Hello world!');
  //     console.log('Hello world!');
  //     console.log('Hello world!');
  //     console.log('Hello world!');
  //
  //     setTimeout(()=>{
  //       console.log('hello vue.js');
  //       console.log('hello vue.js');
  //       console.log('hello vue.js');
  //       console.log('hello vue.js');
  //
  //       setTimeout(()=>{
  //         console.log('hello python');
  //         console.log('hello python');
  //         console.log('hello python');
  //         console.log('hello python');
  //       },1000)
  //     },1000)
  //   },1000)
  // })

  // new Promise((resolve, reject) => {
  //   setTimeout(()=>{
  //     resolve();
  //   });
  // }).then(()=>{
  //   console.log('Hello world!');
  //   console.log('Hello world!');
  //   console.log('Hello world!');
  //   console.log('Hello world!');
  //   console.log('Hello world!');
  //   return new Promise((resolve, reject) => {
  //     setTimeout(()=>{
  //       resolve();
  //     });
  //   })
  // }).then(()=>{
  //   console.log('hello vue.js');
  //   console.log('hello vue.js');
  //   console.log('hello vue.js');
  //   console.log('hello vue.js');
  //   return new Promise((resolve, reject) => {
  //     setTimeout(()=>{
  //       resolve();
  //     });
  //   })
  // }).then(()=>{
  //   console.log('hello python');
  //   console.log('hello python');
  //   console.log('hello python');
  //   console.log('hello python');
  // });

  //什么情况下会用到Promise?
  //一般情况下是有异步操作时，使用Promise对这个异步操作进行封装
  // new ->构造函数(1.保存了一些状态信息 2.执行传入的函数)
  //在执行传入的回调函数时，会传入两个参数，resolve, reject.本身又是函数
  new Promise((resolve, reject) => {
    setTimeout(()=>{
      // 1.成功调用resolve
      // resolve('hello vue.js');

      // 2.失败调用reject
      reject('error message');
    },1000)
  }).then((data)=>{
    console.log(data);
  }).catch((reason)=>{
    console.log(reason);
  });

</script>


</body>
</html>